src/components/User/User.tsx

```diff
@@ -0,0 +1,86 @@
+import { Link } from 'react-router-dom';
+import cn from 'classnames';
+import React from 'react';
+import { Person } from '../../types';
+
+type Props = {
+  person: Person,
+  people: Person[],
+  selectedUser?: Person,
+};
+
+const findRelative = (gender: string, user: Person, users: Person[]) => {
+  if (gender === 'm') {
+    return users.filter(oneUser => oneUser.sex === 'm')
+      .find(oneUser => oneUser.name === user.fatherName);
+  }
+
+  if (gender === 'f') {
+    return users.filter(oneUser => oneUser.sex === 'f')
+      .find(oneUser => oneUser.name === user.motherName);
+  }
+
+  return undefined;
+};
```

Comment:
create the single preparedPersons array, rebase these mechanics to a separate file as a standalone func for preparation